### PR TITLE
crypt: docs: extended description

### PR DESCRIPTION
**Pull request by albertony**
#### What is the purpose of this change?

Extend main description of crypt remote.

There have been several questions in the forum about the basics of the crypt remote, so I have extended the top section of the doc page to try to explain some of the most relevant consepts in one place. Much of the text is taken from various forum posts and other documentation. Perhaps it became a bit too long, and it repeats several of the things that are explained further down the page.. Let me know what you think.

#### Was the change discussed in an issue or in the forum before?

https://github.com/rclone/rclone/pull/4765#issuecomment-730681973
https://forum.rclone.org/t/encryption-at-rest-untrusted-destination-server/20539
https://forum.rclone.org/t/is-was-a-security-auditing-procedure-in-place-for-crypt/450
https://forum.rclone.org/t/how-to-correctly-setup-the-salt-for-the-crypt-remote/4273

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [X] ~I have added tests for all changes in this PR if appropriate.~
- [X] ~I have added documentation for the changes if appropriate.~
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by edwardxml** [comment]
I am glad you tackled this thorny one. Will have a look in the morning from a wording not tech perspective. Ed.

**Comment by albertony** [comment]
Not sure I have tackled it, yet, but its a suggestion and a start at least.. :)

I'm was wondering a bit about the terms "remote" and "backend", in my head "remote" is what you configure in rclone config. Backend is the underlying system behind a remote, usually a cloud service such as dropbox, but also crypt is a backend.

The example section configures a remote with name "remote". I think that can lead to some confusion. It makes it difficult to discuss remote as a general term and this specific configured remote, as in: "the underlying remote is called remote:path". I considered changing this, but it seems this is sort of a "standard" throughout the documentation.

**Comment by edwardxml** [comment]
> start at least.. :)

It is another good bit of progress!

> "remote" and "backend"

I spend quite some time deciding which I should be referring to. Backends seem to be items off the list of backends and remotes are users' instances of them - as you say, as defined in rclone.conf.

> The example section configures a remote with name "remote".

As far as I can tell, that was just done for convenience. Crypt is a bit different because your creation explanation has to cover two remotes. I don't think it needs to be an issue.

A few generl points that struck me:

1. Is it good practice to make the crypt remote a subdirectory of the underlying remote? If so we maybe need to say that.

2. Should the subdirectory being used as crypt empt before being defined as a crypt remote? Do we say that anytwhere. It goes to the heart of us saying that to encrypt files they have to be loaded into a crypt remote. They are not encrypted by just defining a directory containing the files as a crypt.

3. Which leads to changing passwords. Do we need a two para subsection for that? Essemtially the need is to remove the prior encrypted files, change the password etc and then re-upload (or create a fresh crypt directory maybe with use of alias wrap remote). again the point is that changing the password on a remote doesn't change the encryption of pre existing files in a crypt - which is the concept people in the rclone forum seem to have struggled generally with. A subsection might be an opportunity to draw attention to the 1.53.3 disjoint - not often you get the change to do a highlight box!. It isn''t often rclone asks users to break backwards compatibility on their data. I don't know how @ncw has managed it really.

4. You might have removed the section on encrypting local folders / usb removeable drives. I struggled with where that belongs in the page. Maybe again it could have its own subsection. It doesn't particularly fit anywhere else but it does raise an interesting idea that can get the reader thinking about the capabilities and potential uses of rclone, albeit it isn't a mainstream use.

I will try to comment on the actual words in the PR as well today.

Cheers, Ed.

**Comment by albertony** [comment]
Very good points there, thank you. I will look into it in more depth later.
Regarding the local folders / usb, I did not remove the text if that was what you meant? I basically moved it from being a separate section to be a part of the first section, following the text about the flexibility of the feature.

**Comment by edwardxml** [comment]
> I did not remove the text if that was what you meant

Yes, just got to it. Personally I just don't think it belongs in the intro - based on the fact I left it there myself last time! It is a useful oddball point that was made and deserves to be on the page

I was returning to edit/correct my earlier post but you beat me to it :-)

Ed.

**Comment by edwardxml** [comment]
> which is a very strong and highly regarded encryption method

This is something of an opinion and one I have little argument but I don't think it is necessary in reference documentation. This isn't front page sales documentation and to be fair to the average reader of this site they, once presented with the facts as you set them out, are competent to make a good judgement themselves. Those that are not competent to do so will have their own arbitrary beliefs and disregard whatever you say anyway.

> names is optional, and is also weaker

To make this sort of thing shorter you don't need the second 'is' and the 'also' is a bit superfluous given that it follows an 'and'. So 'names is optional but weaker' has the same meaning. Mostly I am trying to strip out excess words and shorten sentences but don't get too exercised about it because that can always be done afterwards - getting the meaning in to start with is what matters.

> content will be encrypted

Mostly I find it clearer to put most things in the present tense and reserve indications something is conditional or set in the future for those few occasions in reference documentation where it matters. So in this, and many other cases, 'content is encrypted' is simpler and has a satisfactorily equivalent meaning.

I think I saw a repeated 'the the' somewhere but you will already have got that!

There is some good new stuff in there.

Ed.

File: docs/content/crypt.md [hunk 1]
```diff
@@ -119,20 +191,79 @@ Rclone does not encrypt
...
+effectively decrypting everything on the fly using the old password and
+re-encrypting using the new password. When done, delete the original crypt
+remote directory and finally the rclone crypt configuration with the old password.
+All data will be streamed from the storage system and back, so you will
+get half the bandwith and be charged twice if you have upload and download quota
+on the storage system.
+
+**Note**: Passwords generated by rclone config in version 1.49.0 (released 2019-08-26)
+to 1.53.2 (released 2020-10-26) are not considered secure and should be changed.
+then it will have been selected from a limited set of passwords and should be changed.
```
Comment (ncw) [inline docs/content/crypt.md:260]: then it will => They will

**Review by ncw — COMMENTED** [review]
I think you've done a brilliant job with that, thank you :-)

You've answered all the questions that come up frequently on the forum (like forgetting the `:` in your remote).

I noticed one minor typo - see inline.

If you can fix that then I'll merge it :-)

**Comment by albertony** [comment]
I gave it a wash and a rebase. If there are any more typos or other comments let me know and I will look into it, but if not I'm fine with it being merged as is.

**Review by ncw — APPROVED** [review]
Looking very good and a great improvement - thank you :-)

I will merge this now.

**Comment by ncw** [comment]
I fixed the conflict.

Will merge now.

**Pull request merged**